model/vm_host.rb

```diff
@@ -67,6 +67,25 @@ def ip6_random_vm_network(prefix = 79)
     end
   end
 
+  def ip4_random_vm_network
+    return nil unless ipv4_bucket
+
+    subnet = NetAddr::IPv4Net.parse(ipv4_bucket)
+    selected_subnet = NetAddr::IPv4Net.new(subnet.nth(SecureRandom.random_number(subnet.len - 1).to_i), NetAddr::Mask32.new(32))
+    
+    ip4_random_vm_network if selected_subnet.network.to_s.nil? || vms.any? { |vm| vm.ip4 == selected_subnet.network.to_s }
+    selected_subnet
+  end
+
+  def vetho_ip4_local_random_vm_network
```

Comment:
much better name needed here. Essentially, we get 1 address here and use 2 consecutive next addresses to assign route for vetho*, vethi* and tap*. So, we are splitting the local address space to three but only store 1 of them for the existence check.